sched/environ/env_dup.c

```diff
@@ -94,28 +96,50 @@ int env_dup(FAR struct task_group_s *group)
         {
           /* There is an environment, duplicate it */
 
-          envp = (FAR char *)group_malloc(group, envlen + 1);
+          envp = group_malloc(group, sizeof(*envp) * (envlen + 1));
           if (envp == NULL)
             {
               /* The parent's environment can not be inherited due to a
                * failure in the allocation of the child environment.
                */
 
-              envlen = 0;
-              ret    = -ENOMEM;
+              ret = -ENOMEM;
             }
           else
             {
               /* Duplicate the parent environment. */
 
-              memcpy(envp, ptcb->group->tg_envp, envlen + 1);
+              do
+                {
+                  if (ptcb->group->tg_envp[envlen] == NULL)
+                    {
+                      envp[envlen] = NULL;
+                      continue;
```

Comment:
optional: below code can be moved to `else` and `continue` can be eliminated. Or even this may be moved out of `do {} while` like:
```
/* Duplicate the parent environment. */

envp[envlen--] = NULL;

for (; envlen > 0; envlen--)
  {
/* perform allocation here */
  }
```

Reply:
Done.